.github/workflows/js-tests.yml

```diff
@@ -118,3 +118,148 @@ jobs:
       - name: Build package
         working-directory: component-lib
         run: yarn build
+
+  js-coverage-comment:
+    needs:
+      - js-unit-tests
+    runs-on: ubuntu-latest
+    permissions:
+      # Required for downloading artifacts
+      actions: read
+      # Required for creating/updating PR comments
+      pull-requests: write
+    if: github.event_name == 'pull_request'
+
+    steps:
+      - name: Checkout Streamlit code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.ref }}
+          persist-credentials: false
+          submodules: "recursive"
+
+      - name: Download coverage JSON
+        uses: actions/download-artifact@v4
+        with:
+          name: vitest_coverage_json
+          path: /tmp/current_coverage
+
+      - name: Get latest develop run ID
+        id: get-latest-run
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const { data: runs } = await github.rest.actions.listWorkflowRuns({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              workflow_id: 'js-tests.yml',
+              branch: 'develop',
+              status: 'success',
+              per_page: 1
+            });
+
+            if (runs.workflow_runs.length === 0) {
+              throw new Error('No successful workflow runs found on develop branch');
+            }
+
+            core.setOutput('run_id', runs.workflow_runs[0].id);
+
+      - name: Download develop coverage
+        uses: actions/download-artifact@v4
+        with:
+          repository: streamlit/streamlit
+          run-id: ${{ steps.get-latest-run.outputs.run_id }}
+          name: vitest_coverage_json
+          path: /tmp/develop_coverage
+          # Needs a token to download the artifact from another
+          # workflow run:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Check coverage changes and comment
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const COVERAGE_CHANGE_THRESHOLD = 0.03; // 0.03%
```

Comment:
Just want to note, that `0.03` looks small number, but in reality since we already have good coverage, it would be really not easy to achieve 😄

Reply:
yeah, I think we can also go a bit lower here, maybe 0.02? I just don't want this to become too annoying :)

Reply:
`0.03` sounds good to me!

Reply:
ok, let's merge and see how the coverage moves over time 👍